### DISCUSSION TITLE
Silly question about Python

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#753] [General] [answered]
Does OpenROAD exist as a package that I can import?

If not, how can I debug Python code, write unit-tests that run from within e.g. Visual Studio Code, combine with other libraries, etc.?

**Comment by vvbandeira** [comment]
There's an ongoing effort to add a python interface to OpenROAD. @macd, do you have any useful pointers or suggestions here?

**Comment by macd** [comment, marked as answer]
We have basic Python wrappers for many of the OR tools, but I think it is still early days for the kind of integration that the OP is looking for. At present, the Python core plus the wrappers are linked into the OR executable. When you execute 'openroad -python', you get the OR specific python interpreter, but you get the system's Python environment (ie available packages etc). In OR python, you can import the packages you want, including pdb. There is a draft PR that seeks to make all this a package and pip installable (https://github.com/The-OpenROAD-Project/OpenROAD/pull/2311), but I don't think there is a timeline for completion. We're not there yet, but the direction is clear.

**Reply by oharboe** [reply]
Sweet!
